samples/v2/lightweight_python_functions_v2_pipeline/lightweight_python_functions_v2_pipeline.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2021 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sample pipeline for passing data in KFP v2."""
+from typing import Dict, List
+
+from kfp import dsl
+from kfp import components
+from kfp.components import InputPath, OutputPath
+from kfp.v2.dsl import Input, Output, Dataset, Model, component
+import kfp.compiler as compiler
+
+
+@component
+def preprocess(
+    # An input parameter of type string.
```

Comment:
nit: would it make sens to move these comments to the docstring instead? https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
In principle I agree to using docstring consistently. However I feel the inlined comments here are more helpful to readers to demonstrate usage. I'd like to hear opinions from other reviewers as well.

Reply:
Overall, I agree on using docstring for component interface description.
But for this specific sample, the comments here are sort of documentations that explain our new annotations: `Input[]`, `Output[]`, and how they compare to the existing annotations like `OutputPath`. 
In that sense, they are more like comments for technical details than docstrings which usually describe the meaning of the function arguments. So I agree with James that this seems to be more helpful to demonstrate usage. WDYT? @NikeNano

Reply:
I prefer to have it in the doc string but I can see the advantage.